Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -0,0 +1,464 @@
+                    for (int i = 0; i < poly.pathCount; ++i)
+                    {
+                        int numPoints = poly.GetPath(i, path);
+                        List<Vector2> dst = new List<Vector2>();
+                        for (int j = 0; j < numPoints; ++j)
+                        {
+                            dst.Add(path[j] * revertCompositeColliderScale);
+                        }
+                        m_originalPath.Add(dst);
+                    }
+                }
+                else
+                {
+                    Invalidate();
+                    return false; // input collider is invalid
+                }
+
+                confinerBaker.BakeConfiner(m_originalPath, aspectRatio, m_bakedConfinerResolution, 
+                    maxOrthoSize, true);
+                m_confinerStates = confinerBaker.GetShrinkablePolygonsAsConfinerStates();
+
+                m_aspectRatio = aspectRatio;
+                m_boundingShape2D = boundingShape2D;
+                m_maxOrthoSize = maxOrthoSize;
+                SetTransformCache(boundingShape2D.transform);
+                CalculateDeltaTransformationMatrix();
+                CalculateOffset();
+
+                return true;
+            }
+
+            private bool IsValid(in Collider2D boundingShape2D, 
+                in float aspectRatio, in float maxOrthoSize)
+            {
+                return boundingShape2D != null && 
+                       m_boundingShape2D != null && m_boundingShape2D == boundingShape2D && // same boundingShape?
+                       //!BoundingShapeTransformChanged(boundingShape2D.transform) && // input shape changed?
+                       m_originalPath != null && // first time?
+                       m_confinerStates != null && // cache not empty? 
+                       Mathf.Abs(m_aspectRatio - aspectRatio) < UnityVectorExtensions.Epsilon && // aspect changed?
+                       Mathf.Abs(m_maxOrthoSize - maxOrthoSize) < UnityVectorExtensions.Epsilon; // max ortho changed?
+            }
+
+            private void SetTransformCache(in Transform boundingShapeTransform)
+            {
+                m_boundingShapeScale = boundingShapeTransform.localScale;
+                m_boundingShapeRotation = boundingShapeTransform.rotation;
+            }
+            
+            /// <summary>
+            /// Transforms point to confiner local space
+            /// </summary>
+            /// <param name="point">Point to transform</param>
+            /// <returns>Point in confiner's local space</returns>
+            public Vector3 TransformPointToConfinerSpace(in Vector3 point)
+            {
+                Vector3 pointInConfinerSpace = T.MultiplyPoint3x4(point);
+                pointInConfinerSpace = R.MultiplyPoint3x4(pointInConfinerSpace);
+                pointInConfinerSpace = S.MultiplyPoint3x4(pointInConfinerSpace);
+                return pointInConfinerSpace;
```

Comment:
I don't understand why you're decomposing the matrix and doing all this kooky math.  Can't you just grab boundingShape.transform.localToWorldMatrix and worldToLocalMatrix?

Reply:
If it's because of the offset then you can incorporate it by premultiplying and getting a new matrix with the combined operations.  If you need help with the matrix math, you can ask me